edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Serializer/MetricsSerializer.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    /// <summary>
+    /// Provides a way to serialize the datetime and value component of a metric.
+    /// </summary>
+    public class RawMetricValue : IEquatable<RawMetricValue>
```

Comment:
Please add tests for RawMetricValue and RawMetric

Reply:
Metric serialization tests test these